From f02c8d741365864a525919c7ad055e28369dbb00 Mon Sep 17 00:00:00 2001
From: Markus Kappeler <markus.kappeler@bytesatwork.ch>
Date: Thu, 5 Mar 2020 14:57:34 +0100
Subject: [PATCH] fdts: stm32mp1-bytedevkit: Introduce bytedevkit board

Add the device tree file to support the bytesatwork's byteDEVKIT.

Signed-off-by: Markus Kappeler <markus.kappeler@bytesatwork.ch>
Signed-off-by: Daniel Ammann <daniel.ammann@bytesatwork.ch>
Signed-off-by: Guy Morand <guy.morand@bytesatwork.ch>
---
 fdts/stm32mp157c-bytedevkit-fw-config.dts     |  7 +++
 .../stm32mp157c-bytedevkit-v1-1-fw-config.dts |  1 +
 fdts/stm32mp157c-bytedevkit-v1-1.dts          |  1 +
 .../stm32mp157c-bytedevkit-v1-3-fw-config.dts |  1 +
 fdts/stm32mp157c-bytedevkit-v1-3.dts          |  1 +
 fdts/stm32mp157c-bytedevkit.dts               | 53 +++++++++++++++++++
 fdts/stm32mp15xx-byteengine.dtsi              | 30 +++++++++++
 7 files changed, 94 insertions(+)
 create mode 100644 fdts/stm32mp157c-bytedevkit-fw-config.dts
 create mode 120000 fdts/stm32mp157c-bytedevkit-v1-1-fw-config.dts
 create mode 120000 fdts/stm32mp157c-bytedevkit-v1-1.dts
 create mode 120000 fdts/stm32mp157c-bytedevkit-v1-3-fw-config.dts
 create mode 120000 fdts/stm32mp157c-bytedevkit-v1-3.dts
 create mode 100644 fdts/stm32mp157c-bytedevkit.dts
 create mode 100644 fdts/stm32mp15xx-byteengine.dtsi

diff --git a/fdts/stm32mp157c-bytedevkit-fw-config.dts b/fdts/stm32mp157c-bytedevkit-fw-config.dts
new file mode 100644
index 000000000..de4dfd0e3
--- /dev/null
+++ b/fdts/stm32mp157c-bytedevkit-fw-config.dts
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) 2022 bytesatwork AG - https://www.bytesatwork.io
+ */
+
+#define DDR_SIZE	0x20000000 /* 512MB */
+#include "stm32mp15-fw-config.dtsi"
diff --git a/fdts/stm32mp157c-bytedevkit-v1-1-fw-config.dts b/fdts/stm32mp157c-bytedevkit-v1-1-fw-config.dts
new file mode 120000
index 000000000..5bcb26d07
--- /dev/null
+++ b/fdts/stm32mp157c-bytedevkit-v1-1-fw-config.dts
@@ -0,0 +1 @@
+stm32mp157c-bytedevkit-fw-config.dts
\ No newline at end of file
diff --git a/fdts/stm32mp157c-bytedevkit-v1-1.dts b/fdts/stm32mp157c-bytedevkit-v1-1.dts
new file mode 120000
index 000000000..871e8ef8f
--- /dev/null
+++ b/fdts/stm32mp157c-bytedevkit-v1-1.dts
@@ -0,0 +1 @@
+stm32mp157c-bytedevkit.dts
\ No newline at end of file
diff --git a/fdts/stm32mp157c-bytedevkit-v1-3-fw-config.dts b/fdts/stm32mp157c-bytedevkit-v1-3-fw-config.dts
new file mode 120000
index 000000000..5bcb26d07
--- /dev/null
+++ b/fdts/stm32mp157c-bytedevkit-v1-3-fw-config.dts
@@ -0,0 +1 @@
+stm32mp157c-bytedevkit-fw-config.dts
\ No newline at end of file
diff --git a/fdts/stm32mp157c-bytedevkit-v1-3.dts b/fdts/stm32mp157c-bytedevkit-v1-3.dts
new file mode 120000
index 000000000..871e8ef8f
--- /dev/null
+++ b/fdts/stm32mp157c-bytedevkit-v1-3.dts
@@ -0,0 +1 @@
+stm32mp157c-bytedevkit.dts
\ No newline at end of file
diff --git a/fdts/stm32mp157c-bytedevkit.dts b/fdts/stm32mp157c-bytedevkit.dts
new file mode 100644
index 000000000..89a8bc479
--- /dev/null
+++ b/fdts/stm32mp157c-bytedevkit.dts
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) 2022 bytesatwork AG - https://www.bytesatwork.io
+ * Author: Markus Kappeler <markus.kappeler@bytesatwork.ch>.
+ */
+
+/dts-v1/;
+
+#include "stm32mp157.dtsi"
+#include "stm32mp15xc.dtsi"
+#include "stm32mp15-pinctrl.dtsi"
+#include "stm32mp15xxac-pinctrl.dtsi"
+#include "stm32mp15xx-byteengine.dtsi"
+#include <dt-bindings/soc/st,stm32-etzpc.h>
+
+/ {
+	model = "byteDEVKIT";
+	compatible = "bytesatwork,bytedevkit", "bytesatwork,byteengine-stm32mp1", "st,stm32mp157";
+
+	aliases {
+		serial0 = &uart4;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+};
+
+&pinctrl {
+	uart4_pins_c: uart4-2 {
+		pins1 {
+			pinmux = <STM32_PINMUX('H', 13, AF8)>; /* UART4_TX */
+			bias-disable;
+			drive-push-pull;
+			slew-rate = <0>;
+		};
+
+		pins2 {
+			pinmux = <STM32_PINMUX('H', 14, AF8)>; /* UART4_RX */
+			bias-disable;
+		};
+	};
+};
+
+&uart4 {
+	pinctrl-names = "default" ;
+	pinctrl-0 = <&uart4_pins_c>;
+	status = "okay";
+};
+
+&cryp1 {
+	status = "okay";
+};
diff --git a/fdts/stm32mp15xx-byteengine.dtsi b/fdts/stm32mp15xx-byteengine.dtsi
new file mode 100644
index 000000000..1fd160830
--- /dev/null
+++ b/fdts/stm32mp15xx-byteengine.dtsi
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) 2024 bytesatwork AG - https://www.bytesatwork.io
+ * Author: Markus Kappeler <markus.kappeler@bytesatwork.ch>
+ */
+
+#include "stm32mp15xx-dkx.dtsi"
+
+&qspi {
+	pinctrl-names = "default";
+	pinctrl-0 = <&qspi_clk_pins_a &qspi_bk1_pins_a>;
+	reg = <0x58003000 0x1000>, <0x70000000 0x4000000>;
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	flash0: w25n01g@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-rx-bus-width = <2>;
+		spi-tx-bus-width = <2>;
+		spi-max-frequency = <50000000>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+	};
+};
+
+&usbotg_hs {
+	usb-role-host;
+};
-- 
2.39.5

